File: android/app/src/main/java/com/example/sparkling/go/DeepLinkHolder.kt
package com.example.sparkling.go

// DeepLinkHolder.kt
object DeepLinkHolder {
    var userId: String = ""
    var email:  String = ""
    var uuid:   String = ""
    var token:  String = ""
    var hasPendingVerify: Boolean = false

    fun set(userId: String, email: String, uuid: String, token: String) {
        this.userId  = userId
        this.email   = email
        this.uuid    = uuid
        this.token   = token
        this.hasPendingVerify = true
    }

    fun consume(): Map<String, String>? {
        if (!hasPendingVerify) return null
        val result = mapOf(
            "userId" to userId,
            "email"  to email,
            "uuid"   to uuid,
            "token"  to token,
        )
        hasPendingVerify = false
        return result
    }
}

File: ios/SparklingGo/SparklingGo/DeepLinkHolder.swift
import Foundation

final class DeepLinkHolder {
    static let shared = DeepLinkHolder()
    private init() {}

    private(set) var userId: String = ""
    private(set) var email: String = ""
    private(set) var uuid: String = ""
    private(set) var token: String = ""
    private(set) var hasPendingVerify: Bool = false

    func set(userId: String, email: String, uuid: String, token: String) {
        self.userId = userId
        self.email = email
        self.uuid = uuid
        self.token = token
        self.hasPendingVerify = true
    }

    func consume() -> [String: String]? {
        guard hasPendingVerify else { return nil }
        let result = [
            "userId": userId,
            "email":  email,
            "uuid":   uuid,
            "token":  token,
        ]
        hasPendingVerify = false
        return result
    }
}
